partialWaveFit/fitResult.cc

```diff
@@ -243,103 +262,188 @@ fitResult::fill(const unsigned int              nmbEvents,               // numb
+                const TMatrixT<double>&         fitParCovMatrix,         // covariance matrix of fit parameters
+                const vector<pair<int, int> >&  fitParCovMatrixIndices,  // indices of fit parameters for real and imaginary part in covariance matrix matrix
+                const complexMatrix&            normIntegral,            // normalization integral matrix
+                const complexMatrix&            acceptedNormIntegral,    // normalization integral matrix with acceptance
+                const vector<double>&           phaseSpaceIntegral,      // normalization integral over full phase space without acceptance
+                const bool                      converged,
+                const bool                      hasHessian)
+{
+	fill(nmbEvents, normNmbEvents, multibinBoundaries, logLikelihood, rank, prodAmps, prodAmpNames,
+	     &fitParCovMatrix, fitParCovMatrixIndices, &normIntegral, &acceptedNormIntegral, &phaseSpaceIntegral,
+	     converged, hasHessian);
+}
+
 
 void
-fitResult::fill(const fitResult& result)
+fitResult::fill(const fitResult& result,
+                const TMatrixT<double>*         fitParCovMatrix,         // covariance matrix of fit parameters
+                const vector<pair<int, int> >*  fitParCovMatrixIndices,  // indices of fit parameters for real and imaginary part in covariance matrix matrix
+                const complexMatrix*            normIntegral,            // normalization integral matrix
+                const complexMatrix*            acceptedNormIntegral,    // normalization integral matrix with acceptance
+                const vector<double>*           phaseSpaceIntegral)      // normalization integral over full phase space without acceptance
 {
-	_nmbEvents              = result.nmbEvents();
-	_normNmbEvents          = result.normNmbEvents();
-	_multibinBoundaries     = result.multibinBoundaries();
-	_logLikelihood          = result.logLikelihood();
-	_rank                   = result.rank();
-	_prodAmps               = result.prodAmps();
-	_prodAmpNames           = result.prodAmpNames();
-	_waveNames              = result.waveNames();
-	_covMatrixValid         = result.covMatrixValid();
-
-	const TMatrixT<double>& fitParCovMatrix = result.fitParCovMatrix();
-	_fitParCovMatrix.ResizeTo(fitParCovMatrix.GetNrows(), fitParCovMatrix.GetNcols());
-	_fitParCovMatrix        = fitParCovMatrix;
-
-	_fitParCovMatrixIndices = result.fitParCovIndices();
-	_normIntegral           = result.normIntegralMatrix();
-	_acceptedNormIntegral   = result.acceptedNormIntegralMatrix();
-	_normIntIndexMap        = result.normIntIndexMap();
-	_phaseSpaceIntegral     = result.phaseSpaceIntegralVector();
-	_converged              = result.converged();
-	_hasHessian             = result.hasHessian();
+	if (fitParCovMatrix == nullptr)
+		fitParCovMatrix = (not result.covMatrixValid()) ? nullptr : &result.fitParCovMatrix();
+	if (fitParCovMatrixIndices == nullptr)
```

Comment:
why is this still a pointer here?

Reply:
Because this function should give you the ability to copy a fit result and overwrite certain matrices.

Reply:
So we will have different types for `fitParCovMatrixIndices` for the various `fill` functions just for a feature (overwriting the mapping of production amplitudes to covariance matrix entries) that is not used anywhere in the current code?

Reply:
This `fill` function, which allows overwriting of matrices, will be used here #238.
Only the `fitParCovMatrixIndices` will not be overwritten, but I think this `fill` function should also have this possibility for completeness.